pymc3/smc/sample_smc.py

```diff
@@ -313,23 +309,16 @@ def sample_smc(
             trace.report._run_convergence_checks(idata, model)
     trace.report._log_summary()
 
-    posterior = idata if return_inferencedata else trace
-    if save_sim_data:
-        return posterior, {modelcontext(model).observed_RVs[0].name: np.array(sim_data)}
-    else:
-        return posterior
+    return idata if return_inferencedata else trace
```

Comment:
when is the trace going to be deprecated? maybe we could only return inferencedata

Reply:
I would only deprecate it when it is deprecated in `pm.sample`